subprojects/docs/src/docs/userguide/managingTransitiveDependencies.adoc

```diff
@@ -17,6 +17,24 @@
 
 Resolution behavior for transitive dependencies can be customized to a high degree to meet enterprise requirements.
 
+[[sec:dependency_constraints]]
+=== Dependency constraints
+
+Dependency constraints allow you to define the version or the version range of both dependencies declared in the build script and transitive dependencies. It is the preferred method to express constraints that should be applied to all dependencies of a configuration.
+
+++++
+<sample id="dependencyConstraints" dir="userguide/dependencies/dependencyConstraints" title="Define dependency constraints">
+    <sourcefile file="build.gradle" snippet="dependency-constraints"/>
+</sample>
+++++
+
+In the example, all versions are omitted from the dependency declaration. Instead, the versions are defined in the constraints block. The version definition for `commons-codec:1.11` is only taken into account if `commons-codec` is brought in as transitive dependency, since `commons-codec` is not defined as dependency in the project. Otherwise, the constraint has no effect.
```

Comment:
Is the result of any version conflict resolution going to be overwritten? Meaning it wins last. How are constraints affected by resolution rules?

Reply:
We should probably go into a bit more detail here. I added a few explaining sentences. Let me know what you think.

Reply:
:o: Thanks for making it clearer. One point I am still missing is how resolution rules are treated e.g. this https://docs.gradle.org/current/userguide/customizing_dependency_resolution_behavior.html#sec:dependency_resolve_rules.

Reply:
Good point. They are applied afterwards. I added a sentence about that.